File: spec/test_parser_identical_rules_lookahead.c
#include "testparser.h"
#include <string.h>
#include <assert.h>

int main()
{
    char const * input = "aba";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    input = "abb";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    return 0;
}

File: spec/test_parser_identical_rules_lookahead.rs
use testparser::*;

fn main() {
    for input in [&b"aba"[..], &b"abb"[..]] {
        let mut context = p_context_new(input);
        assert_eq!(P_SUCCESS, p_parse(&mut context));
        p_context_delete(context);
    }
}
